frontend/app/src/components/history/ClosedTrades.vue

```diff
@@ -242,360 +307,357 @@ enum TradeFilterKeys {
+          action: trade.tradeType,
+          amount: trade.amount
+        })
+        .toString();
+      tradeToDelete.value = trade;
+    };
 
-  setSelected(selected: boolean) {
-    const selection = this.selected;
-    if (!selected) {
-      const total = selection.length;
-      for (let i = 0; i < total; i++) {
-        selection.pop();
+    const deleteTradeHandler = async () => {
+      if (!tradeToDelete.value) {
+        return;
       }
-    } else {
-      for (const trade of this.data) {
-        const { tradeId } = trade;
-        if (!tradeId || selection.includes(tradeId)) {
-          logger.warn(
-            'A problematic trade has been detected, possible duplicate id',
-            trade
-          );
-        }
-        selection.push(tradeId);
-      }
-    }
-  }
 
-  selectionChanged(tradeId: string, selected: boolean) {
-    const selection = this.selected;
-    if (!selected) {
-      const index = selection.indexOf(tradeId);
-      if (index >= 0) {
-        selection.splice(index, 1);
+      const success: boolean = await deleteTrade(tradeToDelete.value?.tradeId);
+
+      if (!success) {
+        return;
       }
-    } else if (tradeId && !selection.includes(tradeId)) {
-      selection.push(tradeId);
-    }
-  }
 
-  get allSelected(): boolean {
-    const strings = this.data.map(({ tradeId }) => tradeId);
-    return (
-      strings.length > 0 && isEqual(sortBy(strings), sortBy(this.selected))
-    );
-  }
+      tradeToDelete.value = null;
+      confirmationMessage.value = '';
+      fetch();
+    };
 
-  async ignoreTrades(ignore: boolean) {
-    let status: ActionStatus;
-
-    const actionIds = this.data
-      .filter(({ tradeId, ignoredInAccounting }) => {
-        return (
-          (ignore ? !ignoredInAccounting : ignoredInAccounting) &&
-          this.selected.includes(tradeId)
-        );
-      })
-      .map(({ tradeId }) => tradeId)
-      .filter((value, index, array) => array.indexOf(value) === index);
-
-    if (actionIds.length === 0) {
-      const choice = ignore ? 1 : 2;
-      this.setMessage({
-        success: false,
-        title: this.$tc('closed_trades.ignore.no_actions.title', choice),
-        description: this.$tc(
-          'closed_trades.ignore.no_actions.description',
-          choice
-        )
-      });
-      return;
-    }
-    const payload: IgnoreActionPayload = {
-      actionIds: actionIds,
-      type: IgnoreActionType.TRADES
+    const clearDialog = () => {
+      const formVal = form.value as any;
+      formVal?.reset();
+
+      openDialog.value = false;
+      editableItem.value = null;
     };
-    if (ignore) {
-      status = await this.ignoreActions(payload);
-    } else {
-      status = await this.unignoreActions(payload);
-    }
 
-    if (status.success) {
-      const total = this.selected.length;
-      for (let i = 0; i < total; i++) {
-        this.selected.pop();
+    const save = async () => {
+      const formVal = form.value as any;
```

Comment:
The type of form is supposed to be `ExtrernalTradeForm | null` correct? Is there any problem that forces us to cast it to any?